File: phalcon_data/app/library/Socket.php
<?php

namespace app\library;

use \WebSocket\Client;

/* WebSocket客户端 */
class Socket{

  static private $client = null;

  /* 发送 */
  static function send($msg){
    // 链接
    if(!self::$client){
      $config = require APP_PATH.'/config/env.php';
      self::$client = new Client('ws://'.$config['socket_ip'].':'.$config['socket_port'].'/?token='.$config['key']);
    }
    // 发送
    self::$client->send(json_encode($msg));
    // echo self::$client->receive();
  }

  /* 发送-cli */
  static function sendCli($msg){
    $config = require APP_PATH.'/config/env.php';
    exec($config['cli'].'socket send \''.json_encode($msg).'\' 2>&1');
  }

}

File: java/src/main/java/vip/webmis/java/library/Socket.java
package vip.webmis.java.library;

import java.net.URI;
import java.util.HashMap;

import com.alibaba.fastjson.JSON;

import org.java_websocket.client.WebSocketClient;
import org.java_websocket.handshake.ServerHandshake;

import vip.webmis.java.Env;

/* WebSocket */
public class Socket extends WebSocketClient {

  /* 发送 */
  public static void send(String type, HashMap<String, Object> msg){
    try {
      String url = "ws://"+Env.socket_ip+":"+Env.socket_port+"/websocket?type=admin&token="+Env.key;
      Socket clients = new Socket(new URI(url));
      clients.connect();
      while (!clients.getReadyState().toString().equals("OPEN")) {
        Thread.sleep(300);
      }
      clients.send(JSON.toJSONString(msg));
    } catch (Exception e) {

    }
  }

  /* 客户端 */
  public Socket(URI serverUri) {
    super(serverUri);
  }
  /* 链接 */
  @Override
  public void onOpen(ServerHandshake arg0) {
    // System.out.println("1、链接");
  }
  /* 消息 */
  @Override
  public void onMessage(String message) {
    // System.out.println("2、消息: "+message);
  }
  /* 关闭 */
  @Override
  public void onClose(int code, String reason, boolean remote) {
    // System.out.println("3、关闭");
  }
  /* 错误 */
  @Override
  public void onError(Exception ex) {
    // System.out.println("4、错误");
  }

}
